code/modules/vehicles/unmanned/unmanned_droid.dm

```diff
@@ -83,3 +96,33 @@
 	playsound(src, 'sound/effects/pred_cloakoff.ogg', 60, TRUE)
 	alpha = initial(alpha)
 	TIMER_COOLDOWN_START(src, COOLDOWN_DRONE_CLOAK, 12 SECONDS)
+
+///Proc used for the supply link feature, activate to appear as an antenna
+/obj/vehicle/unmanned/droid/proc/activate_antenna(datum/source, mob/user)
+	SIGNAL_HANDLER
+
+	for(var/mob/temp_user AS in client_mobs_in_contents)
+		user = temp_user
+
+	if(beacon_datum)
+		UnregisterSignal(beacon_datum, COMSIG_PARENT_QDELETING)
+		QDEL_NULL(beacon_datum)
+		to_chat(user, (span_warning("The [src] beeps and states, \"Your last position is no longer accessible by the supply console")))
+		return
+	if(!is_ground_level(src.z))
+		to_chat(user, span_warning("You have to be on the planet to use this or it won't transmit."))
+		return FALSE
+	beacon_datum = new /datum/supply_beacon(user.name, src.loc, user.faction, 4 MINUTES)
+	RegisterSignal(beacon_datum, COMSIG_PARENT_QDELETING, .proc/clean_beacon_datum)
+	user.show_message(span_notice("The [src] beeps and states, \"Your current coordinates were registered by the supply console. LONGITUDE [loc.x]. LATITUDE [loc.y]. Area ID: [get_area(src)]\""))
+
+/obj/vehicle/unmanned/droid/proc/clean_beacon_datum()
+	SIGNAL_HANDLER
+	beacon_datum = null
+
+/datum/action/antenna
+	name = "Use Antenna"
+	action_icon_state = "rally_minions"
+
+/datum/action/antenna/action_activate()
+	SEND_SIGNAL(owner, COMSIG_UNMANNED_COORDINATES)
```

Comment:
I dont know what to think about this signal

Reply:
Looks fine to me, you cannot really get the droid from the owner